classes/Tools.php

```diff
@@ -4307,6 +4309,40 @@ public static function isFileFresh(string $file, int $timeout = self::CACHE_LIFE
 
         return false;
     }
+
+    /**
+     * @return bool
+     */
+    public static function isDetectCountryFromBrowserLanguageAvailable(): bool
+    {
+        $languagesAvailable = self::languagesAvailable();
+
+        if (!isset($languagesAvailable[0])) {
+            return false;
+        }
+
+        return Configuration::get('PS_DETECT_COUNTRY') && isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])
+            && self::isValidAcceptLanguageAvailable()
+            && Validate::isLanguageIsoCode($languagesAvailable[0]);
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isValidAcceptLanguageAvailable(): bool
```

Comment:
```suggestion
    public static function doesHttpHeaderProvideValidLanguage(): bool
```
This function checks whether the HTTP headers provide a valid language
⬇️ 
isValidAcceptLanguageAvailable

Reply:
```suggestion
    public static function isCountryIsoCodeAvailableInHeader(): bool
```

Reply:
this too, I will apply it